core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -388,10 +388,11 @@ private[spark] class AppStatusListener(
 
       job.completionTime = if (event.time > 0) Some(new Date(event.time)) else None
       update(job, now, last = true)
+      if(job.status.equals(JobExecutionStatus.SUCCEEDED)) {
```

Comment:
Nits: space after `if`. Can we write `job.stats == JobExecutionStatus.SUCCEEDED`? might be more idiomatic.
Certainly the result of the change looks reasonable, but I don't know this code enough to be sure it's the right change. It seems reasonable.